core/types/permissions_cache.go

```diff
@@ -110,77 +110,126 @@ type PermissionConfig struct {
 	SubOrgBreadth *big.Int         `json:"subOrgBreadth"`
 }
 
+var syncStarted = false
+
+var DefaultAccess = FullAccess
+var QIP714BlockReached = false
+var networkAdminRole string
+var orgAdminRole string
+
+const defaultOrgMapLimit = 2000
+const defaultRoleMapLimit = 2500
+const defaultNodeMapLimit = 1000
+const defaultAccountMapLimit = 6000
+
+var OrgInfoMap = NewOrgCache()
+var NodeInfoMap = NewNodeCache()
+var RoleInfoMap = NewRoleCache()
+var AcctInfoMap = NewAcctCache()
+
+
 type OrgKey struct {
 	OrgId string
 }
 
-type NodeKey struct {
-	OrgId string
-	Url   string
+type OrgCache struct {
+	c                 *lru.Cache
+	mux               sync.Mutex
+	evicted           bool
+	populateCacheFunc func(orgId string) *OrgInfo
+}
+
+
+func (o *OrgCache) PopulateCacheFunc(cf func(string) *OrgInfo) {
+	o.populateCacheFunc = cf
+}
+
+func NewOrgCache() *OrgCache {
+	orgCache := OrgCache{evicted: false}
+	onEvictedFunc := func(k interface{}, v interface{}) {
+		orgCache.evicted = true
+	}
+	orgCache.c, _ = lru.NewWithEvict(defaultOrgMapLimit, onEvictedFunc)
```

Comment:
handle error returned by NewWithEvict

Reply:
The only condition when `NewWithEvict` can return error is when the size passed is non zero, which is ensured by passing `defaultOrgMapLimit` which is set to value greater than zero. Hence this error handling is not required.